Libraries/LibWeb/Credentials/NavigatorCredentialsContainer.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGC/Ptr.h>
+#include <LibWeb/Credentials/CredentialsContainer.h>
+
+namespace Web::Credentials {
+class NavigatorCredentialsContainerMixin {
+public:
+    virtual ~NavigatorCredentialsContainerMixin() = default;
+
+    [[nodiscard]] GC::Ref<CredentialsContainer> credentials();
+
+protected:
+    virtual Bindings::PlatformObject const& this_navigator_storage_object() const = 0;
```

Comment:
well. almost :D 

For this API, it would be called `this_navigator_credentials_object()`.

The name of `this_navigator_storage_object` came from the fact that the other mixin was called "navigator storage"